### PR TITLE
{bio}[foss/2016b] HiC-Pro v2.9.0, bx-python v0.7.4

### PR DESCRIPTION
**Pull request by timeu**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/b/bx-python/bx-python-0.7.4-foss-2016b-Python-2.7.12.eb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+# this is a bundle of Python packages
+exts_defaultclass = 'PythonPackage'
+
+prebuildopts = "export CPATH=$EBROOTLZO/include/lzo:$CPATH && "
+
+exts_list = [
+    ('python-lzo', '1.11', {
+        'modulename': 'lzo',
+        'source_urls': ['https://pypi.python.org/packages/source/p/python-lzo/'],
+        'source_tmpl': 'python-lzo-%(version)s.tar.gz',
```
Comment (verdurin) [inline easybuild/easyconfigs/b/bx-python/bx-python-0.7.4-foss-2016b-Python-2.7.12.eb]: @timeu - please add checksums for the extensions with `--inject-checksums`.

File: easybuild/easyconfigs/h/HiC-Pro/HiC-Pro-2.9.0-foss-2016b-Python-2.7.12.eb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+versionsuffix = '-Python-%(pyver)s'
+
+homepage = 'http://nservant.github.io/HiC-Pro'
+description = """HiC-Pro was designed to process Hi-C data, from raw fastq files (paired-end Illumina data) 
+ to the normalized contact maps."""
+
+toolchain = {'name': 'foss', 'version': '2016b'}
+
+sources = ['v%(version)s.tar.gz']
+source_urls = ['https://github.com/nservant/HiC-Pro/archive/']
```
Comment (verdurin) [inline easybuild/easyconfigs/h/HiC-Pro/HiC-Pro-2.9.0-foss-2016b-Python-2.7.12.eb]: @timeu please add an SHA256 checksum.

File: easybuild/easyconfigs/b/bx-python/bx-python-0.7.4-foss-2016b-Python-2.7.12.eb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+version = '0.7.4'
+versionsuffix = '-Python-%(pyver)s'
+
+homepage = 'https://github.com/bxlab/bx-python'
+description = """The bx-python project is a Python library and associated set of scripts to allow for rapid
+ implementation of genome scale analyses."""
+
+toolchain = {'name': 'foss', 'version': '2016b'}
+
+
```
Comment (verdurin) [inline easybuild/easyconfigs/b/bx-python/bx-python-0.7.4-foss-2016b-Python-2.7.12.eb]: @timeu - please remove this unnecessary extra blank line

File: easybuild/easyconfigs/h/HiC-Pro/HiC-Pro-2.9.0-foss-2016b-Python-2.7.12.eb [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+description = """HiC-Pro was designed to process Hi-C data, from raw fastq files (paired-end Illumina data) 
+ to the normalized contact maps."""
+
+toolchain = {'name': 'foss', 'version': '2016b'}
+
+source_urls = ['https://github.com/nservant/HiC-Pro/archive/']
+sources = ['v%(version)s.tar.gz']
+checksums = ['b82f15747bce937c3337b8120ad331a4bc07bbed6708d1d38b7d6ff1047a7a81']
+
+# buildopts=" configure"
```
Comment (verdurin) [inline easybuild/easyconfigs/h/HiC-Pro/HiC-Pro-2.9.0-foss-2016b-Python-2.7.12.eb]: @timeu - any reason why this comment is still here? Something you found during testing?

Comment (timeu) [reply easybuild/easyconfigs/h/HiC-Pro/HiC-Pro-2.9.0-foss-2016b-Python-2.7.12.eb]: no I think it was an artifact from a copy. I will remove it.

File: easybuild/easyconfigs/h/HiC-Pro/HiC-Pro-2.9.0-foss-2016b-Python-2.7.12.eb [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        'source_tmpl': 'v%(version)s.tar.gz',
+        'source_urls': ['https://github.com/hiclib/iced/archive/'],
+        'checksums': ['6a199d713b1d8211c843afa8ec0caa83c640212470e735ebde3729c344537690'],
+    }),
+]
+
+skipsteps = ['configure']
+
+files_to_copy = [
+    (['scripts'], ''),
```
[the remark this answers is not in the document]
Comment (timeu) [reply easybuild/easyconfigs/h/HiC-Pro/HiC-Pro-2.9.0-foss-2016b-Python-2.7.12.eb]: hmm good question, I guess the line below is redundant. I can remove it

[the remark this answers is not in the document]
Comment (boegel) [reply easybuild/easyconfigs/h/HiC-Pro/HiC-Pro-2.9.0-foss-2016b-Python-2.7.12.eb]: You should change this entry to just `scripts` imho...

File: easybuild/easyconfigs/h/HiC-Pro/HiC-Pro-2.9.0-foss-2016b-Python-2.7.12.eb [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+source_urls = ['https://github.com/nservant/HiC-Pro/archive/']
+sources = ['v%(version)s.tar.gz']
+checksums = [
+    'b82f15747bce937c3337b8120ad331a4bc07bbed6708d1d38b7d6ff1047a7a81',  # v2.9.0.tar.gz
+    '7006902cff8367b6bf1ef8cabd4de26bdbfa842062a0250755ff703227e88f34',  # HiC-Pro.patch
+]
+
+exts_defaultclass = 'PythonPackage'
+
+patches = ['HiC-Pro.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/h/HiC-Pro/HiC-Pro-2.9.0-foss-2016b-Python-2.7.12.eb]: @timeu Please rename the patch to something like `HiC-Pro-2.9l0_eb.patch`, and move this up between `sources = ` and `checksums =`

File: easybuild/easyconfigs/h/HiC-Pro/HiC-Pro-2.9.0-foss-2016b-Python-2.7.12.eb [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+        'source_tmpl': 'v%(version)s.tar.gz',
+        'source_urls': ['https://github.com/hiclib/iced/archive/'],
+        'checksums': ['6a199d713b1d8211c843afa8ec0caa83c640212470e735ebde3729c344537690'],
+    }),
+]
+
+skipsteps = ['configure']
+
+files_to_copy = [
+    ('scripts'),
```
Comment (boegel) [inline easybuild/easyconfigs/h/HiC-Pro/HiC-Pro-2.9.0-foss-2016b-Python-2.7.12.eb]: you can drop the `(...)` here?

File: easybuild/easyconfigs/h/HiC-Pro/HiC-Pro-2.9.0-foss-2016b-Python-2.7.12.eb [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+    (['bin/utils', 'bin/HiC-Pro'], 'bin'),
+    (['config-system.txt'], '')
```
Comment (boegel) [inline easybuild/easyconfigs/h/HiC-Pro/HiC-Pro-2.9.0-foss-2016b-Python-2.7.12.eb]: this is equivalent with just using `'config-system.txt'`?

File: easybuild/easyconfigs/h/HiC-Pro/HiC-Pro-2.9.0-foss-2016b-Python-2.7.12.eb [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+# specify that Bundle easyblock should run a full sanity check, rather than just trying to load the module
+full_sanity_check = True
+
+sanity_check_paths = {
+    'files': ['bin/ice', 'config-system.txt', 'scripts/build_matrix', 'scripts/cutsite_trimming', 'bin/HiC-Pro'],
+    'dirs': ['lib/python%(pyshortver)s/site-packages', 'scripts', 'bin/utils'],
+}
+
+postinstallcmds = [
```
Comment (boegel) [inline easybuild/easyconfigs/h/HiC-Pro/HiC-Pro-2.9.0-foss-2016b-Python-2.7.12.eb]: this should be moved up, below `files_to_copy` (order doesn't matter in an easyconfig file, but style-wise we try to keep easyconfig parameter more-or-less in order of execution)

File: easybuild/easyconfigs/h/HiC-Pro/HiC-Pro.patch [hunk 10]
```diff
@@ -0,0 +1,34 @@
+diff -ruN HiC-Pro-2.9.0/config-system.txt HiC-Pro-2.9.0-patched/config-system.txt
```
Comment (boegel) [inline easybuild/easyconfigs/h/HiC-Pro/HiC-Pro.patch]: @timeu Please include a short description of the patch above this line, and mention the author (or include a reference to where you got the patch from)

**Comment by boegel** [comment]
@timeu ping?

**Comment by timeu** [comment]
@boegel Sorry for the delay.  I will take care of the remaining issues today.

File: easybuild/easyconfigs/h/HiC-Pro/HiC-Pro.patch (hunk 10 above)
Comment (boegel) [reply easybuild/easyconfigs/h/HiC-Pro/HiC-Pro.patch]: @timeu Please remove this old (now unused) patch?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2594.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/f03d6e7cf445bce5d9fb791e241861ce for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2015.delcatty.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/ff903bf21223694d9f36ccc5ddcc87cf for a full test report.

**Comment by boegel** [comment]
Going in, thanks @timeu!

**Pull request merged**